icx-proxy/src/logging.rs

```diff
@@ -0,0 +1,79 @@
+use crate::Opts;
+use slog::{Drain, Level, LevelFilter, Logger};
+use std::fs::File;
+use std::path::PathBuf;
+
+/// The logging mode to use.
+enum LoggingMode {
+    /// The default mode for logging; output without any decoration, to STDERR.
+    Stderr,
+
+    /// Tee logging to a file (in addition to STDERR). This mimics the verbose flag.
+    /// So it would be similar to `dfx ... |& tee /some/file.txt
+    Tee(PathBuf),
+
+    /// Output Debug logs and up to a file, regardless of verbosity, keep the STDERR output
+    /// the same (with verbosity).
+    File(PathBuf),
+}
+
+fn create_drain(mode: LoggingMode) -> Logger {
+    match mode {
+        LoggingMode::File(out) => {
+            let file = File::create(out).expect("Couldn't open log file");
+            let decorator = slog_term::PlainDecorator::new(file);
+            let drain = slog_term::FullFormat::new(decorator).build().fuse();
+            Logger::root(slog_async::Async::new(drain).build().fuse(), slog::o!())
+        }
+        // A Tee mode is basically 2 drains duplicated.
+        LoggingMode::Tee(out) => Logger::root(
+            slog::Duplicate::new(
+                create_drain(LoggingMode::Stderr),
+                create_drain(LoggingMode::File(out)),
+            )
+            .fuse(),
+            slog::o!(),
+        ),
+        _ => {
```

Comment:
Suggestion: be explicit about the other case so that a compiler error will result if an entry is added to the enum later, without also adding a handler case here.
```suggestion
        LoggingMode::Stderr => {
```